brickstrap-image-drivers.sh

```diff
@@ -22,6 +23,43 @@
 # brickstrap.
 #
 
+
+#
+# Implement a single root file system partition scheme.
+# This function creates a MBR type image (type extension: img).
+# $1: path to the image file to generate.
+#
+function brp_image_drv_single_fs()
+{
+    debug "IMAGE_FILE_SIZE: ${IMAGE_FILE_SIZE}"
+    debug "ROOT_PART_NAME: ${ROOT_PART_NAME}"
+
+    [ $# -eq 1 -a -n "$1" ] && guestfish -N \
+        "$1"=fs:ext4:${IMAGE_FILE_SIZE} \
+        set-label /dev/sda1 ${ROOT_PART_NAME} : \
+        mount /dev/sda1 / : \
+        tar-in "$(br_tarball_path)" / : \
+
+}
+
+#
+# Sanity checks configuration variables for brp_image_drv_single_fs
+#
+# Variables:
+# ROOT_PART_NAME: Label of root partition. Default: ROOTFS
+# IMAGE_FILE_SIZE: The size of the entire image file. Default: 3800M
+#
+function brp_image_drv_check_single_fs()
+{
+    ROOT_PART_NAME=${ROOT_PART_NAME:-ROOTFS}
+    IMAGE_FILE_SIZE=${IMAGE_FILE_SIZE:-3800M}
+
+    [ ${#ROOT_PART_NAME} -gt 16 ] && \
```

Comment:
It might be preferable to extract this as utility function that takes two argument and returns either 0 or non-zero exit code. `brp_image_check_part_name` or something. Call like this:
```
brp_image_check_part_name ROOT_PART_NAME "$ROOT_PART_NAME"
```
... And update other fs drivers that want the same validation for partition names to take advantage of this as well?